sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkLogger.java

```diff
@@ -30,7 +30,10 @@ class SdkLogger implements Logger {
 
   private final LoggerSharedState loggerSharedState;
   private final InstrumentationScopeInfo instrumentationScopeInfo;
-  private final boolean loggerEnabled;
+
+  // deliberately not volatile because of performance concerns
```

Comment:
I assumed the same considerations hold here as for #7021, let me know if that's not true so I can replace this with an `AtomicBoolean`